expression/builtin_math_vec.go

```diff
@@ -108,6 +108,48 @@ func (b *builtinAsinSig) vectorized() bool {
 	return true
 }
 
+func (b *builtinAtan2ArgsSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf0); err != nil {
+		return err
+	}
+
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	result.ResizeFloat64(n, false)
```

Comment:
we can use the `result` variable to store one of it's parameter's result. Thus the vectors used in the calculation can be reduced from 3 to 2.

Reply:
got it!